src/java.base/windows/native/libnet/NetworkInterface_winXP.c

```diff
@@ -274,20 +267,23 @@ int getAllInterfacesAndAddresses (JNIEnv *env, netif **netifPP)
     // Retrieve IPv4 addresses with the IP Helper API
     curr = *netifPP;
     ret = lookupIPAddrTable(env, &tableP);
-    if (ret < 0) {
+    if (ret == -1) {
+      free_netif(*netifPP);
       return -1;
+    } else if (ret == -2) {
+        // Clear the exception and continue.
+        if ((*env)->ExceptionCheck(env)) {
+            (*env)->ExceptionClear(env);
+        }
+        tableP = NULL;
     }
     while (curr != NULL) {
```

Comment:
Maybe we should not enter here if tableP == null. Looks like it should be `while ( curr != NULL && tableP != null)` - either that or guard the block with an `if (tableP != null)` ?